internal/pkg/gateway/commit/notifier.go

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright 2021 IBM All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package commit
+
+import (
+	commonProto "github.com/hyperledger/fabric-protos-go/common"
+	ordererProto "github.com/hyperledger/fabric-protos-go/orderer"
+	peerProto "github.com/hyperledger/fabric-protos-go/peer"
+	"github.com/hyperledger/fabric/common/ledger/blockledger"
+)
+
+// Notifier enables notification of commits to a peer ledger.
+type Notifier struct {
+	reader BlockReader
+}
+
+// NewNotifier returns a new commit notifier that obtains channels from a given channel factory.
+func NewNotifier(reader BlockReader) *Notifier {
+	return &Notifier{
+		reader: reader,
+	}
+}
+
+// Notify the supplied channel when the specified transaction is committed to a channel's ledger.
+func (notifier *Notifier) Notify(channelID string, transactionID string) (<-chan peerProto.TxValidationCode, error) {
+	// TODO: add a context parameter to enable the notifier to be cancelled
+	// TODO: pooling of ledger iterators over multiple invocations
+	blockIterator, err := notifier.reader.Iterator(channelID, seekNextCommit())
+	if err != nil {
+		return nil, err
+	}
+
+	commitChannel := make(chan peerProto.TxValidationCode, 1)
```

Comment:
Just a note.  I think that returning the channel for the result here is adequate, because in this case, we always expect for `readCommit` to read indefinitely, and never expect an error to occur other than (once wired) the context cancellation.  If `readCommit` were to have an expected error conditions which would cause the channel to be simply closed without populating a result, then I think we would want to use a slightly more sophisticated data structure which could pass an error or return status.  But, I think this is fine for this case.

Reply:
There are currently error paths where the channel can be closed without populating a result. How should I best handle this?